### ISSUE TITLE
Bad MIME type on some 404s

### ISSUE THREAD

**Issue by aredridel** [tootsuite/mastodon#758]
`< Content-Type: #<Mime::NullType:0x007fb1dc8011d8>`

```
:; curl https://mastodon.social/@foo@bar.net -v
*   Trying 51.15.128.226...
* TCP_NODELAY set
* Connected to mastodon.social (51.15.128.226) port 443 (#0)
* TLS 1.2 connection using TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
* Server certificate: mastodon.social
* Server certificate: Let's Encrypt Authority X3
* Server certificate: DST Root CA X3
> GET /@foo@bar.net HTTP/1.1
> Host: mastodon.social
> User-Agent: curl/7.51.0
> Accept: */*
>
< HTTP/1.1 404 Not Found
< Date: Mon, 03 Apr 2017 00:45:56 GMT
< Content-Type: #<Mime::NullType:0x007fb1dc8011d8>
< Transfer-Encoding: chunked
< Connection: keep-alive
< Server: Mastodon
< X-Frame-Options: DENY
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Vary: Accept-Encoding
< Cache-Control: no-cache
< Set-Cookie: _mastodon_session=SFdwVmtBODBuMzhZNTNVR0hEM01VN0JyMVhJeVh3Sml5WDZQeVJzWllqY1YzaXZTVi9sVm4zQ0ZiK2lvWGE3MVBpUWtxa3dKSTJjRS9ML2llZUNGQnAzQXhGeGNDdXl4TWhPYjI0MXMzbGIySFhXUDJ6K3V0anRTRjgyMGwvaWYtLS9td21qRHEyUG1JeWlmbEo1SG5DeFE9PQ%3D%3D--50fbe9aaa5deaca20d0f9fc786b5500aea35353c; path=/; secure; HttpOnly
< X-Request-Id: 4f9a3e5e-e01b-487c-b6a7-7e10cee97d2d
< X-Runtime: 0.010857
<
* Curl_http_done: called premature == 0
* Connection #0 to host mastodon.social left intact
```

**Comment by happycoloredbanana** [comment]
The strange MIME type is caused by the way Rails treats unknown formats in ```respond_to``` + ```format.any``` blocks. I have opened rails/rails#28927 for this.

While investigating, I have found another problem in mastodon's code: in many ```respond_to``` blocks ```format.any``` preceeds specific formats. E.g., [here](https://github.com/tootsuite/mastodon/blob/fdcf884cf719b682e413ee047542861b5a5cf3b7/app/controllers/application_controller.rb#L48) ```format.any``` preceeds ```format.html``` meaning HTML will rendered by the "any" line.
